interfaces/builtin/connectivity_observe.go

```diff
@@ -0,0 +1,156 @@
+# Description: Allow owning the NetworkingStatus bus name on the system bus
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+
+dbus (send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member={Request,Release}Name
+   peer=(name=org.freedesktop.DBus),
+
+dbus (bind)
+   bus=system
+   name="com.ubuntu.connectivity1.NetworkingStatus",
+`)
+
+var connectivityObserveConnectedSlotAppArmor = []byte(`
+# Description: Allow access to NetworkingStatus service.
+
+dbus (receive)
+    bus=system
+    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
+    interface=org.freedesktop.DBus.*,
+`)
+
+var connectivityObserveConnectedPlugAppArmor = []byte(`
+# Description: Allow using NetworkingStatus service.
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to NetworkingStatus service
+dbus (receive, send)
+    bus=system
+    interface=com.ubuntu.connectivity1.NetworkingStatus{,/**}
+    path=/com/ubuntu/connectivity1/NetworkingStatus,
+
+dbus (send)
+    bus=system
+    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
+    interface=org.freedesktop.DBus.*,
+`)
+
+var connectivityObservePermanentSlotDBus = []byte(`
+<policy user="root">
+    <allow own="com.ubuntu.connectivity1.NetworkingStatus"/>
+    <allow send_destination="com.ubuntu.connectivity1.NetworkingStatus"/>
+</policy>
+
+<policy context="default">
+    <deny own="com.ubuntu.connectivity1.NetworkingStatus"/>
+
+    <deny send_destination="com.ubuntu.connectivity1.NetworkingStatus"/>
+</policy>
+
+<limit name="max_replies_per_connection">1024</limit>
+<limit name="max_match_rules_per_connection">2048</limit>
+`)
+
+type ConnectivityObserveInterface struct{}
+
+func (iface *ConnectivityObserveInterface) Name() string {
+	return "connectivity-observe"
+}
+
+func (iface *ConnectivityObserveInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *ConnectivityObserveInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		snippet := []byte(connectivityObserveConnectedPlugAppArmor)
+		old := []byte("###PLUG_SECURITY_TAGS###")
+		var new []byte
+		new = slotAppLabelExpr(slot)
+		snippet = bytes.Replace(snippet, old, new, -1)
+		return snippet, nil
+	}
+	return nil, nil
+}
+
+func (iface *ConnectivityObserveInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus:
+		return connectivityObservePermanentSlotDBus, nil
+	case interfaces.SecurityAppArmor:
+		snippet := []byte(connectivityObservePermanentSlotAppArmor)
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		var new []byte
+		new = slotAppLabelExpr(slot)
+		snippet = bytes.Replace(snippet, old, new, -1)
+		return snippet, nil
```

Comment:
No substitutions here. Please use simply:
```
case interfaces.SecurityAppArmor:
        return connectivityObservePermanentSlotAppArmor, nil
```

How this works is that Permanent policy is what is granted to allow the thing to run at all, and Connected policy is what is granted to allow snaps to communicate. As such, connectivityObservePermanentSlotAppArmor has no substitution rules. connectivityObserveConnectedSlotAppArmor has substitution rules for the peer plug's labels and connectivityObserveConnectedPlugAppArmor has substitution rules for the peer slot's labels.